src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt

```diff
@@ -37,53 +40,79 @@ internal class StandaloneNodeFormula(
 
     override fun provision(): StandaloneStoppedNode {
         logger.info("Setting up $name...")
-
         ssh.newConnection().use { connection ->
             computer.setUp(connection)
-            val unpackedProduct = productDistribution.install(connection, ".")
-            val jiraHome = TaskTimer.time("download Jira home") {
-                jiraHomeSource.download(connection)
-            }
-            replaceDbconfigUrl(connection, "$jiraHome/dbconfig.xml")
-            SetenvSh(unpackedProduct).setup(
-                connection = connection,
-                config = config,
-                gcLog = JiraGcLog(unpackedProduct),
-                jiraIp = ssh.host.ipAddress
+        }
+        val executor = Executors.newCachedThreadPool(
+            ThreadFactoryBuilder()
+                .setNameFormat("install-product-with-dataset-${config.name}-%d")
+                .build()
+        )
+
+        try {
+            val unpackedProductFuture = executor.submit(
+                Callable<String> {
+                    ssh.newConnection().use { connection ->
+                        return@Callable productDistribution.install(connection, ".")
+                    }
+                }
             )
-            connection.execute("echo jira.home=`realpath $jiraHome` > $unpackedProduct/atlassian-jira/WEB-INF/classes/jira-application.properties")
-            connection.execute("echo jira.autoexport=false > $jiraHome/jira-config.properties")
-            downloadMysqlConnector(
-                "https://dev.mysql.com/get/Downloads/Connector-J/mysql-connector-java-5.1.40.tar.gz",
-                connection
+
+            val jiraHomeFuture = executor.submit(
+                Callable<String> {
+                    ssh.newConnection().use { connection ->
+                        return@Callable TaskTimer.time("download Jira home") {
+                            jiraHomeSource.download(connection)
+                        }
+                    }
+                }
             )
-            connection.execute("tar -xzf mysql-connector-java-5.1.40.tar.gz")
-            connection.execute("cp mysql-connector-java-5.1.40/mysql-connector-java-5.1.40-bin.jar $unpackedProduct/lib")
-            AwsCli().download(pluginsTransport.location, connection, target = "$jiraHome/plugins/installed-plugins")
+            val jiraHome = jiraHomeFuture.get()
+            val unpackedProduct = unpackedProductFuture.get()
+            ssh.newConnection().use { connection ->
```

Comment:
We can extract the lambda contents to a private method to reduce nesting and method height.

Reply:
@wyrzyk WDYT?

Reply:
> @wyrzyk WDYT?

I could refactor the method. OTOH we're going to replace it with a new API soon, so I don't see much value here.

Reply:
It's less readable than before the PR, so I raised it.
OTOH, it's a small thing, so I will unblock you.